### PR TITLE
no more vars in Web/API (7)

### PR DESCRIPTION
**Pull request by lionralfs**
#### Summary

Replaces `var` in Web/API with `const` where applicable and `let` otherwise.
Mostly automated via my ESLint setup, using the rules `no-var` and `prefer-const`

#### Motivation

* See #16662
* See https://github.com/mdn/content/discussions/10389
* See https://github.com/orgs/mdn/discussions/143

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/wheelevent/deltamode/index.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ An `unsigned long`.
 ## Examples
 
 ```js
-var syntheticEvent = new WheelEvent("syntheticWheel", {"deltaX": 4, "deltaMode": 0});
+const syntheticEvent = new WheelEvent("syntheticWheel", {"deltaX": 4, "deltaMode": 0});
```
Comment (teoli2003) [inline files/en-us/web/api/wheelevent/deltamode/index.md:33]:
```suggestion
const syntheticEvent = new WheelEvent("syntheticWheel", { "deltaX": 4, "deltaMode": 0 });
```

File: files/en-us/web/api/wheelevent/deltax/index.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ A number.
 ## Examples
 
 ```js
-var syntheticEvent = new WheelEvent("syntheticWheel", {"deltaX": 4, "deltaMode": 0});
+const syntheticEvent = new WheelEvent("syntheticWheel", {"deltaX": 4, "deltaMode": 0});
```
Comment (teoli2003) [inline files/en-us/web/api/wheelevent/deltax/index.md:27]:
```suggestion
const syntheticEvent = new WheelEvent("syntheticWheel", { "deltaX": 4, "deltaMode": 0 });
```

File: files/en-us/web/api/wheelevent/deltay/index.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ A number.
 ## Examples
 
 ```js
-var syntheticEvent = new WheelEvent("syntheticWheel", {"deltaY": 4, "deltaMode": 0});
+const syntheticEvent = new WheelEvent("syntheticWheel", {"deltaY": 4, "deltaMode": 0});
```
Comment (teoli2003) [inline files/en-us/web/api/wheelevent/deltay/index.md:27]:
```suggestion
const syntheticEvent = new WheelEvent("syntheticWheel", { "deltaY": 4, "deltaMode": 0 });
```

File: files/en-us/web/api/wheelevent/deltaz/index.md [hunk 4]
```diff
@@ -25,7 +25,7 @@ A number.
 ## Examples
 
 ```js
-var syntheticEvent = new WheelEvent("syntheticWheel", {"deltaZ": 4, "deltaMode": 0});
+const syntheticEvent = new WheelEvent("syntheticWheel", {"deltaZ": 4, "deltaMode": 0});
```
Comment (teoli2003) [inline files/en-us/web/api/wheelevent/deltaz/index.md:28]:
```suggestion
const syntheticEvent = new WheelEvent("syntheticWheel", { "deltaZ": 4, "deltaMode": 0 });
```

File: files/en-us/web/api/window/navigator/index.md [hunk 5]
```diff
@@ -26,7 +26,8 @@ The {{domxref("navigator")}} object.
 ### Example #1: Browser detect and return a string
 
 ```js
-var sBrowser, sUsrAg = navigator.userAgent;
+let sBrowser;
+let sUsrAg = navigator.userAgent;
```
Comment (teoli2003) [inline files/en-us/web/api/window/navigator/index.md:30]:
```suggestion
const sUsrAg = navigator.userAgent;
```

File: files/en-us/web/api/xmlhttprequest/abort/index.md [hunk 6]
```diff
@@ -47,7 +47,7 @@ This example begins loading content from the MDN home page, then due to some con
 aborts the transfer by calling `abort()`.
 
 ```js
-var xhr = new XMLHttpRequest(),
+const xhr = new XMLHttpRequest(),
     method = "GET",
     url = "https://developer.mozilla.org/";
```
Comment (teoli2003) [inline files/en-us/web/api/xmlhttprequest/abort/index.md:52]:
```suggestion
const xhr = new XMLHttpRequest();
const method = "GET";
const url = "https://developer.mozilla.org/";
```

File: files/en-us/web/api/xmlhttprequest/responseurl/index.md [hunk 7]
```diff
@@ -19,7 +19,7 @@ The read-only **`XMLHttpRequest.responseURL`** property returns the serialized U
 ## Example
 
 ```js
-var xhr = new XMLHttpRequest();
+const xhr = new XMLHttpRequest();
 xhr.open('GET', 'http://example.com/test', true);
 xhr.onload = function () {
```
Comment (teoli2003) [inline files/en-us/web/api/xmlhttprequest/responseurl/index.md:24]:
```suggestion
xhr.onload = () => {
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/window/scrolly/index.md [hunk 8]
```diff
@@ -69,11 +69,11 @@ Explorer (< 9) do not support either property and must be worked around by check
 other non-standard properties. A fully compatible example:
 
 ```js
-var supportPageOffset = window.pageXOffset !== undefined;
-var isCSS1Compat = ((document.compatMode || "") === "CSS1Compat");
+const supportPageOffset = window.pageXOffset !== undefined;
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/window/scrolly/index.md]: I removed the whole work-around.

File: files/en-us/web/api/xmlhttprequest/using_xmlhttprequest_in_ie6/index.md [hunk 9]
```diff
@@ -11,18 +11,19 @@ tags:
...
 ```
 
 However, if you need to also support Internet Explorer 6 and older, you need to extend your code like this:
 
 ```js
+let request;
 if (window.XMLHttpRequest) {
     //Firefox, Opera, IE7, and other browsers will use the native object
-    var request = new XMLHttpRequest();
+    request = new XMLHttpRequest();
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/xmlhttprequest/using_xmlhttprequest_in_ie6/index.md:23]: I think we should delete it and redirect to /web/api/xmlhttprequest page (in a separate PR)

File: files/en-us/web/api/window/scrollx/index.md [hunk 10]
```diff
@@ -61,11 +61,11 @@ For cross-browser compatibility, use `window.pageXOffset` instead of
 non-standard properties. A fully compatible example:
 
 ```js
-var x = (window.pageXOffset !== undefined)
+const x = (window.pageXOffset !== undefined)
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/window/scrollx/index.md]: Removed.

File: files/en-us/web/api/xmlhttprequest/using_xmlhttprequest_in_ie6/index.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (lionralfs) [reply files/en-us/web/api/xmlhttprequest/using_xmlhttprequest_in_ie6/index.md:23]: Leaving it as-is for now.

**Review by teoli2003 — APPROVED** [review]

**Pull request merged**